sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemSchema.java

```diff
@@ -244,7 +244,10 @@ public TableType getJdbcTableType()
                   val.getVersion(),
                   val.getShardSpec().getPartitionNum(),
                   partialSegmentData == null ? 0L : partialSegmentData.getNumReplicas(),
-                  partialSegmentData == null ? 0L : partialSegmentData.getNumRows(),
+                  partialSegmentData == null ? 0L
```

Comment:
I think i'd be in favor of moving this outside and assigning variables instead of the 4 repeated checks of `partialSegmentData == null`, e.g.
```
      long numReplicas, numRows, isRealtime = 0L, isAvailable = 1L;
      if (partialSegment != null) {
...
      }
```

Reply:
Made `numRows` of type `long` and refactored the code with @clintropolis 's suggestion. I do not see any other nullable metric, actually now `numRows` is also not nullable.